### PR TITLE
Disable TokenUtilsEncryptTest#testFailAlgorithm on Java 17+

### PR DESCRIPTION
**Pull request by famod**
Relates to https://github.com/quarkusio/quarkus/issues/18372#issuecomment-877864662

A fix upstream would obviously be much nicer than this XML manipulation. 😉

### CODE REVIEW

**Comment by sberyozkin** [comment]
@famod, missed your comment, sorry,

> 2021-07-09T02:41:37.1781961Z Expected exception of type class org.jose4j.jwt.consumer.InvalidJwtException but got org.jose4j.lang.InvalidAlgorithmException: A128KW is an unknown, unsupported or unavailable alg algorithm (not one of [RSA1_5, RSA-OAEP, RSA-OAEP-256, dir, ECDH-ES, A128GCMKW, A192GCMKW, A256GCMKW]). 

Interesting, this is a strange error because something like `dir` is not an algorithm  known to Java (`dir` - is just  a direct encryption without wrapping the generated encryption key). 

Let me look at it a bit more

**Comment by famod** [comment]
@sberyozkin thanks for taking a closer look!
Unless this is fixable quickly, I suggest we merge this in a timely manner because there should be no negative side effects and it can be reverted easily later.

File: tcks/microprofile-jwt/pom.xml [hunk 1]
```diff
@@ -164,4 +164,44 @@
...
+                                    <goal>transform</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                        <configuration>
+                            <transformationSets>
+                                <transformationSet>
+                                    <dir>${project.build.directory}/tck-suite/suites</dir>
+                                    <outputDir>${project.build.directory}/tck-suite/suites</outputDir>
+                                    <stylesheet>src/test/xml/exlusions-jdk17.xsl</stylesheet>
```
Comment (gsmet) [inline tcks/microprofile-jwt/pom.xml:194]:
```suggestion
                                    <stylesheet>src/test/xml/exclusions-jdk17.xsl</stylesheet>
```

File: tcks/microprofile-jwt/src/test/xml/exlusions-jdk17.xsl [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <xsl:output method="xml" doctype-system="http://testng.org/testng-1.0.dtd"/>
+
+    <!-- Identity transform -->
+    <xsl:template match="@* | node()">
+        <xsl:copy>
+            <xsl:apply-templates select="@* | node()"/>
+        </xsl:copy>
+    </xsl:template>
+
+    <!-- Exlude TokenUtilsEncryptTest#testFailAlgorithm due to https://github.com/quarkusio/quarkus/issues/18372#issuecomment-877864662 -->
```
Comment (gsmet) [inline tcks/microprofile-jwt/src/test/xml/exlusions-jdk17.xsl:13]:
```suggestion
    <!-- Exclude TokenUtilsEncryptTest#testFailAlgorithm due to https://github.com/quarkusio/quarkus/issues/18372#issuecomment-877864662 -->
```

File: tcks/microprofile-jwt/src/test/xml/exlusions-jdk17.xsl [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (gsmet) [inline tcks/microprofile-jwt/src/test/xml/exlusions-jdk17.xsl:1]: This file should be renamed (see above).

**Review by gsmet — DISMISSED** [review]
Spotted some minor issues. I agree we should merge it.

File: tcks/microprofile-jwt/pom.xml (hunk 1 above)
Comment (famod) [reply tcks/microprofile-jwt/pom.xml]: 🤦

**Comment by sberyozkin** [comment]
@famod, @gsmet I don't think it should be dropped

It is not dropped but the name has changed. Rather than merging it as is, I'd suggest to add a BC dependency for this case

https://bugs.openjdk.java.net/browse/JDK-8248268.

I'll open a Jose4j issue (to support `AES/KW/NoPadding`)

**Review by sberyozkin — CHANGES_REQUESTED** [review]
I propose to investigate an option to use a BC library for now instead of ignoring the test and also open a specific issue to make sure it is run without BC in due time

**Comment by famod** [comment]
well, I did try adding:
```
                <groupId>org.bouncycastle</groupId>
                <artifactId>bcprov-jdk15on</artifactId>
```
in test scope but AFAICS, this didn't change anything, maybe because it has to be "installed" explicitly?

**Comment by sberyozkin** [comment]
@famod Can you please try passing a system property `quarkus.security.security-providers=BC` (and also use a normal scope for the dep, as I'm not sure the test one is sufficient) ?

**Comment by famod** [comment]
@sberyozkin unfortunately, the following didn't work:
```diff
diff --git a/tcks/microprofile-jwt/pom.xml b/tcks/microprofile-jwt/pom.xml
index b12da6ace4..d9d393d8ad 100644
--- a/tcks/microprofile-jwt/pom.xml
+++ b/tcks/microprofile-jwt/pom.xml
@@ -24,6 +24,7 @@
                     <systemPropertyVariables>
                         <!-- Disable quarkus optimization -->
                         <quarkus.arc.remove-unused-beans>false</quarkus.arc.remove-unused-beans>
+                        <quarkus.security.security-providers>BC</quarkus.security.security-providers>
                         <mp.jwt.tck.jwks.baseURL>http://localhost:8081/</mp.jwt.tck.jwks.baseURL>
                     </systemPropertyVariables>
                     <!-- This workaround allows us to run a single test using
@@ -122,6 +123,11 @@
             </exclusions>
         </dependency>

+        <dependency>
+            <groupId>org.bouncycastle</groupId>
+            <artifactId>bcprov-jdk15on</artifactId>
+        </dependency>
+
         <!-- Minimal test dependencies to *-deployment artifacts for consistent build order -->
         <dependency>
             <groupId>io.quarkus</groupId>
```
Might be related to testng, checking...

**Comment by famod** [comment]
Settings system properties like this should just work for TestNG, so that's not it.

**Comment by sberyozkin** [comment]
@famod OK, let me drop my change request, thanks, perhaps BC does not provide its own implementation. I'll just follow up with a Jose4J issue

**Review by sberyozkin — APPROVED** [review]

**Comment by sberyozkin** [comment]
FYI, https://bitbucket.org/b_c/jose4j/issues/189

**Comment by famod** [comment]
Ok, thanks! I'll create a housekeeping issue to clean this up eventually.

**Pull request merged**